### PR TITLE
refactor: remove `any` from code typing

### LINKED ISSUE #1354: Let's improve ambiguous types more clearly
**Is your feature request related to a problem? Please describe.**
기능 요청이 특정 문제에 연관된 것이라면 여기에 그 문제를 설명해주세요.
There's so many types are `any` in this project. `any` type reduces code stability and interferes type inferencing.
If type information is provided, developers could write more safe code from bugs and issues.

screenshot below is the result of searching for keyword ": any" in this project.

image

**Describe the solution you'd like**
어떤 기능이 있으면 좋겠는지 자세히 설명해주세요.

**Describe alternatives you've considered**
혹시 다른 대안들을 생각해본 적이 있다면 함께 적어주세요.

**Additional context**
기능 요청에 대해 보다 잘 이해할 수 있는 다른 맥락을 기술해주세요.

### PR DESCRIPTION
**Pull request by inureyes**
This PR resolves #1354

### CODE REVIEW

File: src/components/backend-ai-storage-list.ts [hunk 1]
```diff
@@ -1128,11 +1150,12 @@ export default class BackendAiStorageList extends BackendAIPage {
 
   _modifySharedFolderPermissions() {
     const selectNodeList = this.shadowRoot?.querySelectorAll('#modify-permission-dialog wl-select');
-    const inputList = Array.prototype.filter.call(selectNodeList, (pulldown, idx) => pulldown.value !== (this.invitees as any)[idx].perm)
+    console.log(this.invitees);
```
Comment (agatha197) [inline src/components/backend-ai-storage-list.ts]: please remove this debug line.

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]
@inureyes  Can you handle this too? Maybe, `any` -> `Promise[]`
https://github.com/lablup/backend.ai-webui/blob/main/src/components/backend-ai-storage-list.ts#L1924

File: src/components/backend-ai-storage-list.ts (hunk 1 above)
Comment (inureyes) [reply src/components/backend-ai-storage-list.ts]: Resolved.

**Review by Sujin-Kim1 — APPROVED** [review]
LGTM

File: src/components/backend-ai-app-launcher.ts [hunk 2]
```diff
@@ -1237,27 +1237,30 @@ export default class BackendAiAppLauncher extends BackendAIPage {
...
-          tmpInputElement.type = 'text';
-          tmpInputElement.value = copyText;
-
-          document.body.appendChild(tmpInputElement);
-          tmpInputElement.select();
-          document.execCommand('copy'); // copy operation
-          document.body.removeChild(tmpInputElement);
+      const passStore = this.shadowRoot?.querySelector(vscodePassword);
+      if (passStore && 'value' in passStore) {
+        const copyText = passStore.value as string;
```
Comment (yomybaby) [inline src/components/backend-ai-app-launcher.ts]: we can set the type explicit without `as`
```suggestion
      const passStore = this.shadowRoot?.querySelector<HTMLInputElement>(vscodePassword);
      if (passStore && 'value' in passStore) {
        const copyText = passStore.value;
```

File: src/components/backend-ai-credential-view.ts [hunk 3]
```diff
@@ -448,8 +448,10 @@ export default class BackendAICredentialView extends BackendAIPage {
   _launchResourcePolicyDialog() {
     Promise.allSettled([this._getAllStorageHostsInfo(), this._getResourcePolicies()]).then((res) => {
       // TODO remove protected properties usage
-      (this.newPolicyNameInput as any).mdcFoundation.setValid(true);
-      (this.newPolicyNameInput as any).isUiValid = true;
+      // @ts-ignore
+      (this.newPolicyNameInput as TextField).mdcFoundation.setValid(true);
+      // @ts-ignore
+      (this.newPolicyNameInput as TextField).isUiValid = true;
```
Comment (yomybaby) [inline src/components/backend-ai-credential-view.ts]: It's already TextField type. Let's remove `as TextField).
```suggestion
      this.newPolicyNameInput.mdcFoundation.setValid(true);
      // @ts-ignore
      this.newPolicyNameInput.isUiValid = true;
```

File: src/components/backend-ai-data-view.ts [hunk 4]
```diff
@@ -44,7 +44,15 @@ import {IronFlex, IronFlexAlignment, IronPositioning} from '../plastics/layout/i
  * This type definition is a workaround for resolving both Type error and Importing error.
  */
 type BackendAIDialog = HTMLElementTagNameMap['backend-ai-dialog'];
-
+interface groupData {
```
Comment (yomybaby) [inline src/components/backend-ai-data-view.ts]: It's better to start with a capital letter.
```suggestion
interface GroupData {
```

File: src/components/backend-ai-data-view.ts [hunk 5]
```diff
@@ -431,7 +439,7 @@ export default class BackendAIData extends BackendAIPage {
             </mwc-select>
             ${this.is_admin && this.allowed_folder_type.includes('group') ? html`
               <mwc-select class="fixed-position" id="add-folder-group" ?disabled=${this.folderType==='user'} label="${_t('data.Project')}" FixedMenuPosition>
-                ${(this.allowedGroups as any).map((item, idx) => html`
+                ${(this.allowedGroups as groupData[]).map((item, idx) => html`
```
Comment (yomybaby) [inline src/components/backend-ai-data-view.ts]: After setting type of property, we can remove `as ...`:
```typescript
@property({type: Array}) allowedGroups:GroupData[] = [];
```

```suggestion
                ${this.allowedGroups.map((item, idx) => html`
```

File: src/components/backend-ai-session-launcher.ts [hunk 6]
```diff
@@ -2959,11 +2960,10 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
...
 
   _disableEnterKey() {
-    this.shadowRoot?.querySelectorAll('wl-expansion').forEach((element) => {
-      // TODO remove protected property assignment
-      (element as any).onKeyDown = (e) => {
-        const enterKey = 13;
-        if (e.keyCode === enterKey) {
+    this.shadowRoot?.querySelectorAll<Expansion>('wl-expansion').forEach((element) => {
+      // remove protected property assignment
+      (element as HTMLElement).onkeydown = (e) => {
```
Comment (yomybaby) [inline src/components/backend-ai-session-launcher.ts]: We can remove `as ..`
```suggestion
      element.onkeydown = (e) => {
```

**Review by yomybaby — CHANGES_REQUESTED** [review]

File: src/components/backend-ai-session-launcher.ts (hunk 6 above)
Comment (inureyes) [reply src/components/backend-ai-session-launcher.ts]: resolved.

File: src/components/backend-ai-data-view.ts (hunk 5 above)
Comment (inureyes) [reply src/components/backend-ai-data-view.ts]: resolved

File: src/components/backend-ai-data-view.ts (hunk 4 above)
Comment (inureyes) [reply src/components/backend-ai-data-view.ts]: resolved

File: src/components/backend-ai-credential-view.ts (hunk 3 above)
Comment (inureyes) [reply src/components/backend-ai-credential-view.ts]: resolved

File: src/components/backend-ai-app-launcher.ts (hunk 2 above)
Comment (inureyes) [reply src/components/backend-ai-app-launcher.ts]: resolved

**Review by yomybaby — APPROVED** [review]
LGTM!
I push one commit to remove the remaining unnecessary `as` in the other file!

**Pull request merged**